### PR TITLE
migrate to ansible-runner

### PR DESCRIPTION
**Pull request by evgeni**
this fixes ansible 2.8 problems too

### CODE REVIEW

**Comment by evgeni** [comment]
I need to test if record mode still works!

**Comment by evgeni** [comment]
narrator: it does not

**Comment by evgeni** [comment]
narrator: the issues resolved themself after a good lunch

File: test/test_crud.py [hunk 1]
```diff
@@ -46,39 +42,38 @@ def run_playbook_vcr(module, extra_vars=None, extra_args=None, record=False):
     # Assemble extra parameters for playbook call
     if extra_args is None:
         extra_args = []
```
Comment (ekohl) [inline test/test_crud.py]: This is now unused and can be removed

Comment (evgeni) [reply test/test_crud.py]: fair, nuked all extra_vars

File: test/test_crud.py [hunk 2]
```diff
@@ -42,43 +38,39 @@
...
+    path = os.path.join(os.getcwd(), 'test/test_playbooks/{}.yml'.format(module))
+    run = ansible_runner.run(extravars=extra_vars, playbook=path, limit=limit, verbosity=4, inventory=os.path.join(os.getcwd(), 'test/inventory/hosts'))
+    return run
 
 
 @pytest.mark.parametrize('module', MODULES)
 def test_crud(module, record):
-    assert run_playbook_vcr(module, record=record) == 0
+    run = run_playbook_vcr(module, record=record)
+    print(run.stdout.read())
```
Comment (mdellweg) [inline test/test_crud.py]: Is this like file.read() ? Will it read until a buffer is filled?

File: test/test_crud.py [hunk 3]
```diff
@@ -42,43 +38,39 @@
...
-    if extra_vars:
-        playbook_opts.extend(['--extra-vars', extra_vars])
-    if extra_args:
-        playbook_opts.extend(extra_args)
-
-    cli = PlaybookCLI(playbook_opts)
-    cli.parse()
-    return cli.run()
+    os.environ['ANSIBLE_CONFIG'] = os.path.join(os.getcwd(), 'ansible.cfg')
+    path = os.path.join(os.getcwd(), 'test/test_playbooks/{}.yml'.format(module))
```
Comment (ekohl) [inline test/test_crud.py]:
```suggestion
    path = os.path.join(os.getcwd(), 'test', 'test_playbooks', '{}.yml'.format(module))
```

File: test/test_crud.py (hunk 2 above)
Comment (evgeni) [reply test/test_crud.py]: `run()` is blocking, `read()` is like a file, it should read everything.

Comment (evgeni) [reply test/test_crud.py]: https://ansible-runner.readthedocs.io/en/latest/python_interface.html#runner-stdout

**Comment by ehelms** [comment]
Nitpick on changing the commit message to `migrate to ansible-runner for tests`

**Comment by evgeni** [comment]
> Nitpick on changing the commit message to `migrate to ansible-runner for tests`

I can fix that in the squash :)

File: test/test_crud.py [hunk 4]
```diff
@@ -42,43 +38,39 @@
...
-        playbook_opts.extend(['--extra-vars', extra_vars])
-    if extra_args:
-        playbook_opts.extend(extra_args)
-
-    cli = PlaybookCLI(playbook_opts)
-    cli.parse()
-    return cli.run()
+    os.environ['ANSIBLE_CONFIG'] = os.path.join(os.getcwd(), 'ansible.cfg')
+    path = os.path.join(os.getcwd(), 'test', 'test_playbooks', '{}.yml'.format(module))
+    run = ansible_runner.run(extravars=extra_vars, playbook=path, limit=limit, verbosity=4, inventory=os.path.join(os.getcwd(), 'test/inventory/hosts'))
```
Comment (ekohl) [inline test/test_crud.py]: Maybe also change the inventory here to properly use `os.path.join`?

Comment (evgeni) [reply test/test_crud.py]: done

**Pull request merged**